test-data/unit/check-generics.test

```diff
@@ -852,7 +852,8 @@ fun1(1) # E: Argument 1 to "fun1" has incompatible type "int"; expected "List[Tu
 fun1([(1, 'x')]) # E: Cannot infer type argument 1 of "fun1"
 
 reveal_type(fun2([(1, 1)], 1)) # E: Revealed type is 'builtins.list[Tuple[builtins.int*, builtins.int*]]'
-fun2([('x', 'x')], 'x') # E: Value of type variable "T" of "fun2" cannot be "str"
+fun2([('x', 'x')], 'x') # E: Argument types for type variable "T" are incompatible in call to "fun2" \
+                        # N: Arguments "v" and "scale" in call to "fun2" must all have the same type for "T" (one of "int", "bool")
```

Comment:
In understand why you formulate the note this way (because of, e.g.,  ``T`` and ``List[T]`` argument types). But maybe you can show the note only if there are three or less constraint so that in this case you could propose valid variants, like:
```
note: Acceptable combinations of types for arguments 1 and 2 of "fun":
note:    "str" and "List[str]"
note:    "bytes" and "List[bytes]"
note: Got: "bytes" and "List[str]"
```

Reply:
This feels like it might be an enhancement on top of this change (which should end up displaying what we expected to see, and what was actually passed in).